components/script/dom/html/htmlscriptelement.rs

```diff
@@ -1093,7 +1093,7 @@ impl HTMLScriptElement {
         // TODO use a settings object rather than this element's document/window
```

Comment:
nit(non-blocking): I wonder if we should address this TODO and implement [scripting enabled on Global](https://html.spec.whatwg.org/multipage/#concept-environment-noscript)?

It seems that the only difference between the check on Document vs. Global is that the Document check will return false when it does not have a browsing context, and I'm not sure whether there may be some scenarios when that may come into consideration 🤷‍♂️